### DISCUSSION TITLE
JS breakpoints

### DISCUSSION THREAD

**Discussion by velddev** [chakra-ui/panda#1126] [Q&A] [answered]
I'm migrating my project from ChakraUI to Panda CSS, but I'm running into a major issue where I no longer feel like I have a useBreakpointValue, what is the preferred migration method for this?

**Comment by astahmer** [comment]
you can use Panda [`conditions`](https://panda-css.com/docs/concepts/responsive-design) with breakpoints to achieve the same result with CSS only

**Reply by velddev** [reply]
Yea, I tried this, for some context I have a stepper component and a carousel that integrate with each other.

So my problem here is that I would need to show e.g. a specific number of steps based on the number of items in the carousel.

on mobile it could show 1, but on desktop 3. If there are 12 items, it would have 12 pages on mobile and 9 (or 4 if you scroll by threes) on desktop.

I can't really get a value back from the condition to then use in the js runtime. (maybe this is possible? I haven't found how to do it though.)

**Reply by astahmer** [reply, marked as answer]
you can access your DS tokens at runtime with JS using the [`token` function](https://panda-css.com/docs/theming/usage#vanilla-js)
if you need something reactive you can use something like this [`use-breakpoint`](https://github.com/iiroj/use-breakpoint)

**Reply by velddev** [reply]
👍 That's what I ended up doing, thanks for letting me know about the token function!
